lightshow/postprocess/broaden.py

```diff
@@ -83,276 +81,243 @@ def voigt_broaden(x, xin, yin, sigma, gamma, shift=0.0):
+                xout,
+                xin,
+                yin,
+                *broadening_args,
+                ef=self._e_fermi,
+                **broadening_kwargs,
+            )
+        else:
+            raise ValueError(f"Uknown method {method}")
+        self._broadening_kwargs = broadening_kwargs
+        return res
 
-    def paras_optimize(
+    def optimize_parameters(
         self,
-        x,
-        site_spectra,
         exp,
-        bounds=None,
-        opt_shift=False,
-        dmu=True,
-        cross_sec=True,
+        bounds,
+        method="energy_dependent_voigt_broaden",
     ):
-        """Optimize the broadening parameter with simplicial homology global
-        optimization in scipy package.
-
+        """Optimizes the broadening parameters for a single spectrum. A
+        reference spectrum (``exp``) is provided as the ground truth.
+        
         Parameters
         ----------
-        x : numpy.ndarray
-            The output energy grid.
-        site_spectra : list
-            The list of site_spectrum objects.
         exp : numpy.ndarray
-            The experimental spectrum.
-        bounds : list
-            The bounds of the broadening parameters.
-        opt_shift : bool
-            If True, the method will optimize energy shift value.
-        dmu : bool
-            If True, the method will compare the first derivatives of
-            experimental spectrum and simulation.
-        cross_sec : bool
-            If true, the spectrum will be converted to be absorption cross
-            section.
-
-        Returns
-        -------
-        numpy.ndarray
+            Description
+        bounds : None, optional
+            Note that shift is always the first parameter.
+        method : str, optional
+            Description
+        
         """
 
-        if opt_shift is False:
-            xi = 10e-3
+        xin_theory = self._spectrum[:, 0].copy()
 
-            bounds[2] = (self._shift - xi, self._shift + xi)
+        # Bounds[0][0] is negative
+        assert bounds[0][0] <= 0.0
+        lower_shift = exp[0, 0] + bounds[0][0]
+        upper_shift = exp[-1, 0] + bounds[0][1]
 
-        result = optimize.shgo(
-            self._broaden_score,
-            bounds,
-            args=(x, site_spectra, exp, dmu, cross_sec),
-        )
+        self.align_to_experiment_(exp)  # makes the shift optimization easier
 
-        self._sigma = result.x[0]
-        self._lorentz_divider = result.x[1]
+        where_lower = np.argmin((xin_theory - lower_shift)**2)
+        where_upper = np.argmin((xin_theory - upper_shift)**2)
 
-        if opt_shift is True:
-            self._shift = result.x[2]
+        xin_theory = xin_theory[where_lower:where_upper]
+
+        def objective(params):
+            shift = params[0]
+            broadened_spectrum = self.broaden(xin_theory, method, *params[1:])
+            ius = InterpolatedUnivariateSpline(
```

Comment:
@jiangxc66 This probably will work, I just wanted to keep the shifting out of the broadening since shifting the spectral location feels like it shouldn't be in the broadening function, but you're probably right that this will be faster.